### ISSUE TITLE
Support perf enhancements to cs-stream library

### ISSUE THREAD

**Issue by elnyry** [mojaloop/project#1357]
## **Goal**:
As a OSS Developer I want to test this change https://github.com/mojaloop/central-services-stream/pull/94 in performance environment

**Tasks**:
- [x] test and release the change
- [x] test and release for following services 
  - [x] central-ledger
  - [x] ml-api-adapter
  - [x] simulated-message-handler
- [x] run test on the latest perf deployment for data comparison 
- [x] run test in the perf environment with the updated services
- [x] test flow mode
  - [x] implement pause and resume logic, because all messages are consumed in service memory and memory is slowly getting filled.
  - [ ] ~test changes until we have stable behavior~
- [x] test recursive mode
  - [x] implement pause and resume logic, because all messages are consumed in service memory and memory is slowly getting filled.
  - [ ] ~test changes until we have stable behavior~

**Acceptance Criteria**:
- [ ] End-to-end tests are ran and results are captured

**Follow up story**:
- https://github.com/mojaloop/project/issues/1397

## **Accountability**:
- Owner: TBC
- QA/Review: TBC

**Comment by vgenev** [comment]
After testing and documenting the patch from the change, we introduced few more changes:
* async_chained_mode - which waits for the job that the service has to do with a consumed message, before returning and consuming the next message or batch.
* recursive guard to guard against unnecessary recursive calls.

We've started profiling the other modes of operation - flow, poll and recursive. If the changes are more performant and stable, they will be introduced to master branch.
